syntax = "proto3";

package todos;

import "todoService/protos/todo_messages.proto";

service TodoService {
    rpc AddTodo (AddTodoRequest) returns (Todo);
    rpc EditTodo (EditTodoRequest) returns (Todo);
    rpc GetTodo (TodoId) returns (Todo);  // âœ… Uses TodoId
    rpc ListTodos (Empty) returns (ListTodosResponse);
    rpc RemoveTodo (RemoveTodoRequest) returns (RemoveTodoResponse);
}

message ListTodosResponse {
  repeated Todo todos = 1;
}

message RemoveTodoResponse {
  bool success = 1;
}